### PR TITLE
Constant fold int literals

### PR DESCRIPTION
**Pull request by garlic0x1**
https://github.com/coalton-lang/coalton/issues/1176

```lisp
COALTON-USER> (coalton-codegen
                (declare addr (Integer -> Integer))
                (define (addr x) (+ x 1)))
```

Old output, notice the `fromint`
```lisp
(COMMON-LISP:PROGN
 (COALTON-IMPL/GLOBAL-LEXICAL:DEFINE-GLOBAL-LEXICAL ADDR
                                                    COALTON-IMPL/RUNTIME/FUNCTION-ENTRY:FUNCTION-ENTRY)
 (COMMON-LISP:DEFUN ADDR (X-3769)
   (COMMON-LISP:DECLARE (COMMON-LISP:IGNORABLE X-3769))
   (COALTON-LIBRARY/MATH/NUM::|INSTANCE/NUM INTEGER-COALTON-LIBRARY/CLASSES:+|
    X-3769
    (COALTON-LIBRARY/MATH/NUM::|INSTANCE/NUM INTEGER-COALTON-LIBRARY/CLASSES:FROMINT|
     1)))
 (COMMON-LISP:SETF ADDR (COALTON-IMPL/RUNTIME/FUNCTION-ENTRY::F1 #'ADDR))
 (COMMON-LISP:SETF (COMMON-LISP:DOCUMENTATION 'ADDR 'COMMON-LISP:VARIABLE)
                     "ADDR :: (INTEGER → INTEGER)")
 (COMMON-LISP:SETF (COMMON-LISP:DOCUMENTATION 'ADDR 'COMMON-LISP:FUNCTION)
                     "ADDR :: (INTEGER → INTEGER)")
 (COMMON-LISP:VALUES))
```

New output, known integer is not cast.
```lisp
(COMMON-LISP:PROGN
 (COALTON-IMPL/GLOBAL-LEXICAL:DEFINE-GLOBAL-LEXICAL ADDR
                                                    COALTON-IMPL/RUNTIME/FUNCTION-ENTRY:FUNCTION-ENTRY)
 (COMMON-LISP:DEFUN ADDR (X-3770)
   (COMMON-LISP:DECLARE (COMMON-LISP:IGNORABLE X-3770))
   (COALTON-LIBRARY/MATH/NUM::|INSTANCE/NUM INTEGER-COALTON-LIBRARY/CLASSES:+|
    X-3770 1))
 (COMMON-LISP:SETF ADDR (COALTON-IMPL/RUNTIME/FUNCTION-ENTRY::F1 #'ADDR))
 (COMMON-LISP:SETF (COMMON-LISP:DOCUMENTATION 'ADDR 'COMMON-LISP:VARIABLE)
                     "ADDR :: (INTEGER → INTEGER)")
 (COMMON-LISP:SETF (COMMON-LISP:DOCUMENTATION 'ADDR 'COMMON-LISP:FUNCTION)
                     "ADDR :: (INTEGER → INTEGER)")
 (COMMON-LISP:VALUES))
```

Trailing whitespace was removed from the affected file as well.

### CODE REVIEW

**Review by jbouwman — APPROVED** [review]
unambiguous improvement

also thank you for the whitespace cleanup

File: src/codegen/translate-expression.lisp [hunk 1]
```diff
@@ -135,30 +135,36 @@ Returns a `node'.")
...
-         :rands (list
-                 (resolve-dict num-pred ctx env)
-                 (make-node-literal
-                  :type tc:*integer-type*
-                  :value (tc:node-integer-literal-value expr)))))))
+             (from-int-method (util:find-symbol "FROMINT" classes-package))
+
+             (ty (tc:qualified-ty-type qual-ty)))
+
+        (if (and (tc:tycon-p ty) (eq 'coalton:integer (tc:tycon-name ty)))
```
Comment (eliaslfox) [inline src/codegen/translate-expression.lisp:150]: Please add a comment here to explain this optimization.

**Review by eliaslfox — APPROVED** [review]

**Comment by stylewarning** [comment]
Can we get IFix, UFix, single and double floats in this PR too? Those get much greater benefit.

**Comment by stylewarning** [comment]
😑

**Comment by jbouwman** [comment]
🥺

**Pull request merged**